public/service/message.go

```diff
@@ -501,6 +501,20 @@ func (b MessageBatch) TryInterpolatedString(index int, i *InterpolatedString) (s
 	return i.expr.String(index, msg)
 }
 
+// TryInterpolatedAny resolves an interpolated string expression on a message
+// batch, from the perspective of a particular message index.
+//
+// This method allows interpolation functions to perform windowed aggregations
+// across message batches, and is a more powerful way to interpolate strings
+// than the standard .String method.
+func (b MessageBatch) TryInterpolatedAny(index int, i *InterpolatedString) (any, error) {
```

Comment:
I hope this is a reasonable addition. While it might be a bit tricky to troubleshoot if the expression returns an unexpected type, I think it's too cumbersome to have a separate field for the type and then parse the string returned by interpolation to match that type. For Patch operations, users might want to add an object or array field and the API doesn't allow me to pass those as strings.

Reply:
Removed this and switched to full bloblang mapping fields as discussed during the Blobtalk.